cmd/relay-gossip/main.go

```diff
@@ -144,23 +157,40 @@ var clientCmd = &cli.Command{
 			return xerrors.Errorf("constructing host: %w", err)
 		}
 
-		c, err := client.NewWithPubsub(ps, nil, nil)
+		chainHash, err := hex.DecodeString(cctx.String("chain-hash"))
 		if err != nil {
-			return xerrors.Errorf("constructing client: %w", err)
+			return xerrors.Errorf("decoding chain hash: %w", err)
 		}
 
-		var notifChan <-chan drand.PublicRandResponse
-		var unsub client.UnsubFunc
-		{
-			ch := make(chan drand.PublicRandResponse, 5)
-			notifChan = ch
-			unsub = c.Sub(ch)
+		httpFailover := cctx.StringSlice("http-failover")
+
+		var c client.Watcher
+		// if we have http failover endpoints then use the drand HTTP client with pubsub option
+		if len(httpFailover) > 0 {
+			grace := cctx.Duration("http-failover-grace")
+			if grace == 0 {
+				grace = time.Second * 5
+			}
+			c, err = client.New(
```

Comment:
can we make an `[]client.Option` with appropriate options for the two cases, and only create the Client once?

Reply:
No, if you don't specify http failover URLs it doesn't use the drand client.